packages/front-end/components/Experiment/Templates/TemplateForm.tsx

```diff
@@ -0,0 +1,354 @@
+      },
+      type: initialValue?.type ?? "standard",
+      hypothesis: initialValue?.hypothesis || "",
+      description: initialValue?.description || "",
+      tags: initialValue?.tags || [],
+      datasource: initialValue?.datasource || "",
+      userIdType: initialValue?.userIdType || undefined,
+      exposureQueryId: initialValue?.exposureQueryId || "",
+      activationMetric: initialValue?.activationMetric || "",
+      hashAttribute: initialValue?.hashAttribute || hashAttribute,
+      disableStickyBucketing: initialValue?.disableStickyBucketing ?? false,
+      goalMetrics: initialValue?.goalMetrics || [],
+      secondaryMetrics: initialValue?.secondaryMetrics || [],
+      guardrailMetrics: initialValue?.guardrailMetrics || [],
+      statsEngine: initialValue?.statsEngine || orgStatsEngine,
+      skipPartialData: initialValue.skipPartialData ? "strict" : "loose",
+      segment: initialValue.segment || "",
+      targeting: {
+        coverage: initialValue.targeting?.coverage || 1,
+        savedGroups: initialValue.targeting?.savedGroups || [],
+        prerequisites: initialValue.targeting?.prerequisites || [],
+        condition: initialValue.targeting?.condition || "",
+      },
+    },
+  });
+
+  const datasource = form.watch("datasource")
+    ? getDatasourceById(form.watch("datasource") ?? "")
+    : null;
+
+  const { apiCall } = useAuth();
+
+  const onSubmit = form.handleSubmit(async (rawValue) => {
+    const value: ExperimentTemplateInterface = {
+      ...rawValue,
+      templateMetadata: {
+        ...rawValue.templateMetadata,
+        name: rawValue.templateMetadata.name?.trim(),
+      },
+      skipPartialData: rawValue.skipPartialData === "strict",
+    };
+
+    // Make sure there's an experiment name
+    if ((value.templateMetadata?.name?.length ?? 0) < 1) {
+      setStep(0);
+      throw new Error("Template Name must not be empty");
+    }
+
+    // Turn phase dates into proper UTC timestamps
+    validateSavedGroupTargeting(value.targeting?.savedGroups);
+
+    validateAndFixCondition(value.targeting?.condition, (condition) => {
+      form.setValue("targeting.condition", condition);
+      forceConditionRender();
+    });
+
+    if (prerequisiteTargetingSdkIssues) {
+      throw new Error("Prerequisite targeting issues must be resolved");
+    }
+
+    const body = JSON.stringify(value);
+
+    const res = initialValue.id
+      ? await apiCall<{ template: ExperimentTemplateInterface }>(
+          `/templates/${initialValue.id}`,
+          {
+            method: "PUT",
+            body,
+          }
+        )
+      : await apiCall<{ template: ExperimentTemplateInterface }>("/templates", {
+          method: "POST",
+          body,
+        });
+
+    track("Create Experiment Template", {
+      source,
+      numTags: value.tags?.length || 0,
+      numMetrics:
+        (value.goalMetrics?.length || 0) +
+        (value.secondaryMetrics?.length || 0),
+    });
```

Comment:
See what else might be useful to include here. Add whether or not an experiment was created from a template to new experiment tracking event